### DISCUSSION TITLE
section082のkvs_test.goについて

### DISCUSSION THREAD

**Discussion by stutkhd-0709** [budougumi0617/go_todo_app#93] [Ideas]
Redisにキーが存在しない場合NotFoundでアサートしてますが、RedisはNotFoundではなくRedis.nilを返すようです
(github.com/go-redis/redis/v8 v8.11.5)
該当コード
https://github.com/budougumi0617/go_todo_app/blob/61b3a6b88e7501a38023fddca3f131060f1d54fe/store/kvs_test.go#L61-L63

go-redis
https://github.com/redis/go-redis/blob/233f97accde15bcb2e3d86ee744109b4dce34171/string_commands.go#L51

**Comment by budougumi0617** [comment]
ご指摘ありがとうございます！
該当のテストはgo-redisのテストではなく、 `store.KVS` 構造体のテストなので `store.NotFound` が返ってきますね。
`redis. ErrNotFound ` ではなく `ErrNotFound` になっているので同一pkg内で定義された宣言だとわかります。

https://github.com/budougumi0617/go_todo_app/blob/61b3a6b88e7501a38023fddca3f131060f1d54fe/store/kvs.go#L32-L38

CIでもテストはpassしているので。もしテストがfailするならば他の原因があると思います！
https://github.com/budougumi0617/go_todo_app/actions/workflows/test.yml

**Comment by stutkhd-0709** [comment]
あ、なるほど！
```
return 0, fmt.Errorf("failed to get by %q: %w", key, ErrNotFound) 
```
こちらのエラーにErrNotFoundが抜けていたためでした！ありがとうございます！

**Reply by budougumi0617** [reply]
解決してよかったです！また不明点あればお気軽にご記載ください！（確認・返信タイミングはまちまちです 🙇
